scripts/globals/spells/black/drain_ii.lua

```diff
@@ -66,6 +66,17 @@ spellObject.onSpellCast = function(caster, target, spell)
 
     caster:addHP(dmg)
 
+    -- Messaging Fixes
+    local currHP = caster:getHP()
+    local maxHP  = caster:getMaxHP()
+
+    if caster:getHP() == caster:getMaxHP() then
+        spell:setMsg(xi.msg.basic.MAGIC_DRAIN_HP, 0) -- Drains 0 HP
+        return 0
```

Comment:
Sorry to ping on this so much, if you have full HP/MP drain will deal 0 damage?

Reply:
correct. I'll hop into retail to double check this

Reply:
and it's never a bother!

Reply:
I'm actually finding I was previously wrong and you were right! if I have 5 HP to recover, the message will only state how much I recover, not the damage. Moreover full HP and casting drain still deals damage to mob! 
#quickmafs

Reply:
Oh that's great to hear; I thought I was experiencing a Mandela effect that deeply affected long-term memories

Reply:
Solved! Added your suggestion, but also needed to return the min between hpDiff and actually dmg. Dmg is applied separate from this return, so dmg isn't being affected.